comfyui_custom_scripts/extensions/webuiRequests.js

```diff
@@ -110,11 +99,57 @@ function patchSavingMechanism() {
         saveButton.click();
         app.graph.serialize = app.graph.patched_serialize;
     };
-    setTimeout(() => fetch('/webui_scripts/sd-webui-comfyui/default_workflows/postprocess.json')
-        .then(response => response.json())
-        .then(data => {
-            app.loadGraphData(data);
-        }), POLLING_TIMEOUT);
 }
 
-onElementDomIdRegistered(longPolling);
\ No newline at end of file
+async function patchDefaultGraph(workflowTypeId) {
+    const response = await api.fetchApi("/sd-webui-comfyui/default_workflow?" + new URLSearchParams({
+        workflow_type_id: workflowTypeId,
+    }), {
+        method: "GET",
+        headers: {
+            "Content-Type": "application/json",
+        },
+        cache: "no-store",
+    });
+    const defaultGraph = await response.json();
+    if (!defaultGraph) {
+        return;
+    }
+
+    app.original_loadGraphData = app.loadGraphData;
+    app.loadGraphData = (graphData) => {
+        if (!graphData) {
+            return app.original_loadGraphData(defaultGraph);
+        } else {
+            return app.original_loadGraphData(graphData);
+        }
+    };
+
+    app.loadGraphData();
+}
+
+export const iframeRegisteredEvent = new Promise(resolve => {
```

Comment:
do you think it would be a better idea to specify that this is an exported module component at the end of the file? 

```js
export {
    iframeRegisteredEvent,
}
```

Reply:
like that, it becomes easier to track what is and isn't exported, and it's easier to modify in one place

Reply:
We could also move it to a different file and import it instead maybe? This one and `appGraphEvent` below.